task.go

```diff
@@ -81,21 +83,38 @@ func RunTask(name string) error {
 		return &taskNotFoundError{name}
 	}
 
-	for _, d := range t.Deps {
-		if err := RunTask(d); err != nil {
-			return err
-		}
-	}
-
 	if !Force && isTaskUpToDate(t) {
 		log.Printf(`task: Task "%s" is up to date`, name)
 		return nil
 	}
-
+	vars, err := t.handleVariables()
+	if err != nil {
+		return &taskRunError{name, err}
+	}
+  
+	for _, d := range t.Deps {
+		if err := RunTask(ReplaceVariables(d, vars)); err != nil {
+			return err
+		}
+	}
+  
 	for _, c := range t.Cmds {
-		if err := runCommand(c, t.Dir); err != nil {
+		// read in a each time, as a command could change a variable or it has been changed by a dependency
+		vars, err = t.handleVariables()
+		if err != nil {
 			return &taskRunError{name, err}
 		}
+		var (
+			output string
+			err    error
+		)
+		if output, err = runCommand(ReplaceVariables(c, vars), ReplaceVariables(t.Dir, vars)); err != nil {
+			return &taskRunError{name, err}
+		}
+		fmt.Println(output)
```

Comment:
Thought about a quiet switch, but that would imply more, I will change code accoring to your suggestion